src/Seq.App.Opsgenie/OpsgenieApiClient.cs

```diff
@@ -50,21 +48,41 @@ public async Task<HttpResponseMessage> CreateAsync(OpsgenieAlert alert)
                 "application/json");
 
             var response = await _httpClient.PostAsync(OpsgenieCreateAlertUrl, content);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            var result = new OpsGenieResult
+            {
+                StatusCode = (int) response.StatusCode,
+                Ok = response.IsSuccessStatusCode,
+                HttpResponse = response,
+                ResponseBody = responseBody
+            };
+
+            try
+            {
+                var opsGenieResponse =
+                    JsonSerializer.Deserialize<OpsGenieResponse>(responseBody,
+                        new JsonSerializerOptions {PropertyNameCaseInsensitive = true});
 
-            if (!response.IsSuccessStatusCode)
+                result.Response = opsGenieResponse;
+            }
+            catch (Exception ex)
```

Comment:
Any reason not to just let the exception bubble up, here, and be handled by the hosting environment? In general, Seq apps shouldn't customize error handling, since the app host does this consistently for all hosted apps and in future we might make improvements up at that layer.

Reply:
I think the perspective was to allow a consistent approach to error handling based on a signal targeted to the specific appid or appinstanceid (or instance name) that included the context/properties around what the alert was. So for example if we fail to connect to OpsGenie to send the alert, we can send an email or Teams or JIRA alert that preserves the context of what we were alerting, by having an error event that preserves this.

Reply:
I'm still inclined to have this caught for the reasons above. It's fairly serious if an alert doesn't go, and if we can catch what the alert was, then we have opportunity to send an alert via other alerting apps. Overall this speaks to Seq reliability - being able to do fallbacks etc. Unfortunately the hosting environment doesn't preserve the event data ... I suppose it could still throw after this?